tests/playwright/src/specs/podman-machine-resources.spec.ts

```diff
@@ -0,0 +1,208 @@
+ ***********************************************************************/
+
+import type { Locator } from '@playwright/test';
+
+import { CreateMachinePage } from '../model/pages/create-machine-page';
+import { PodmanMachineDetails } from '../model/pages/podman-machine-details-page';
+import { ResourceConnectionCardPage } from '../model/pages/resource-connection-card-page';
+import { ResourcesPage } from '../model/pages/resources-page';
+import { expect as playExpect, test } from '../utility/fixtures';
+import {
+  createPodmanMachineFromCLI,
+  deletePodmanMachine,
+  deletePodmanMachineFromCLI,
+  handleConfirmationDialog,
+} from '../utility/operations';
+import { isLinux, isWindows } from '../utility/platform';
+import { waitForPodmanMachineStartup } from '../utility/wait';
+
+const DEFAULT_PODMAN_MACHINE = 'Podman Machine';
+const DEFAULT_PODMAN_MACHINE_VISIBLE = 'podman-machine-default';
+const RESOURCE_NAME = 'podman';
+let dialog: Locator;
+
+const machineTypes = [
+  {
+    PODMAN_MACHINE_NAME: 'podman-machine-rootless',
+    MACHINE_VISIBLE_NAME: 'Podman Machine rootless',
+    isRoot: false,
+    userNet: false,
+  },
+  {
+    PODMAN_MACHINE_NAME: 'podman-machine-rootful',
+    MACHINE_VISIBLE_NAME: 'Podman Machine rootful',
+    isRoot: true,
+    userNet: false,
+  },
+  {
+    PODMAN_MACHINE_NAME: 'podman-machine-user-networking',
+    MACHINE_VISIBLE_NAME: 'Podman Machine user-networking',
+    isRoot: true,
+    userNet: true,
+  },
+];
+
+for (const { PODMAN_MACHINE_NAME, MACHINE_VISIBLE_NAME, isRoot, userNet } of machineTypes) {
+  test.skip(
+    isLinux || process.env.TEST_PODMAN_MACHINE !== 'true',
+    'Tests suite should not run on Linux platform or if TEST_PODMAN_MACHINE is not true',
+  );
+
+  test.skip(
+    PODMAN_MACHINE_NAME === 'podman-machine-user-networking' && !isWindows,
+    'Testing user networking machine only on Windows',
+  );
+
+  test.beforeAll(async ({ runner, welcomePage, page }) => {
+    runner.setVideoAndTraceName(`${PODMAN_MACHINE_NAME}-resources-e2e`);
+    process.env.KEEP_TRACES_ON_PASS = 'true';
+
+    await welcomePage.handleWelcomePage(true);
+    await waitForPodmanMachineStartup(page);
+    dialog = page.getByRole('dialog', { name: 'Podman', exact: true });
+  });
+
+  test.afterAll(async ({ runner, page }) => {
+    test.setTimeout(120_000);
+
+    if (test.info().status === 'failed') {
+      await deletePodmanMachineFromCLI(PODMAN_MACHINE_NAME);
+      await createPodmanMachineFromCLI();
+    }
+
+    try {
+      await handleConfirmationDialog(page, 'Podman', true, 'Yes');
+      await handleConfirmationDialog(page, 'Podman', true, 'OK');
+    } catch (error) {
+      console.log('No handling dialog displayed', error);
+    }
+
+    await runner.close();
+  });
+
+  test.describe
+    .serial(`${MACHINE_VISIBLE_NAME} Resources creation Verification`, () => {
+      test('Check data for available Podman Machine and stop machine', async ({ page, navigationBar }) => {
```

Comment:
please move to beforeall, so the parametrized tests does not need to always stop the default machine, it is enough to do it in beforeAll.